### DISCUSSION TITLE
Implementing Tables interface for types with optional missing/nothing

### DISCUSSION THREAD

**Discussion by xvries** [JuliaData/Tables.jl#333] [Q&A]
Hey all,

I would like to implement the Tables interface for my custom types. We have a lot of types which have optional fields: Sometimes they have a value and sometimes they report a `nothing` or a `missing`. I guess I could get it to work by custom defining every struct with optional fields, but is there a more simple way. Here is a simple example:

```
struct MyType
	i::Int64
	f::Float64
	o::Union{Nothing, String}
end
Tables.istable(::Type{MyType}) = true
Tables.columnaccess(::Type{MyType}) = true
Tables.columns(m::MyType) = m

m = MyType(1, 0.3, nothing)
DataFrame(m)
```

This does not work because `length(::Nothing)` is not defined. Is there a general way to fix this?

**Comment by quinnj** [comment]
Hmmm, is `MyType` really a "column table" though? It seems more like one instance of a `MyType` object is more like a single row in a row-oriented table? Individual columns in column-tables should be some kind of `AbstractVector`.

**Reply by xvries** [reply]
Indeed if its one element its not really a column table, but if it is a `Vector{MyType}` then I want to be able to make a table out of it. Is that possible in some other way then?
